frappe/utils/pdf.py

```diff
@@ -206,7 +206,8 @@ def read_options_from_html(html):
 
 	toggle_visible_pdf(soup)
 
-	# use regex instead of soup-parser
+	valid_styles = get_print_format_styles(soup)
```

Comment:
Could return a `{name: value}` dictionary from here and then `options = {name: value for name, value in valid_styles.items() if name in attrs}` instead of the nested loop maybe

Reply:
I mainly went with the loop for readability but I don't have a strong opinion here. Used dict comprehension instead 😄